tests/CiIntegrationTest.php

```diff
@@ -0,0 +1,187 @@
+
+        $result2 = static::executeScript(array(
+            $steps[0],
+            $steps[1],
+            $steps[2],
+            'echo "${#EXTRA_ARGS[@]}"',
+            'echo "${EXTRA_ARGS[@]}"',
+            'echo "${EXTRA_ARGS[0]}"',
+            'echo "${EXTRA_ARGS[1]}"',
+            'echo "${EXTRA_ARGS[2]}"',
+            'echo "${EXTRA_ARGS[3]}"',
+        ), true);
+
+        $this->assertSame(
+            array(
+                '4',
+                '--path-mode=intersection -- dir a/file.php dir b/file b.php',
+                '--path-mode=intersection',
+                '--',
+                'dir a/file.php',
+                'dir b/file b.php',
+            ),
+            explode("\n", rtrim($result2['output']))
+        );
+
+        $result3 = static::executeScript(array(
+            $steps[0],
+            $steps[1],
+            $steps[2],
+            $steps[3],
+        ), true);
+
+        $optionalIncompatibilityWarning = 'PHP needs to be a minimum version of PHP 5.3.6 and maximum version of PHP 7.1.*.
+Ignoring environment requirements because `PHP_CS_FIXER_IGNORE_ENV` is set. Execution may be unstable.
+';
+        $executionDetails = 'Loaded config default from ".php_cs.dist".
+S.
+Legend: ?-unknown, I-invalid file syntax, file ignored, S-Skipped, .-no changes, F-fixed, E-error';
+
+        $this->assertRegExp(
+            '/^('.preg_quote($optionalIncompatibilityWarning, '/').')?'.preg_quote($executionDetails, '/').'$/',
+            trim($result3['stderr'])
+        );
+        $this->assertRegExp(
+            '/^Checked all files in \d+\.\d+ seconds, \d+\.\d+ MB memory used$/',
+            trim($result3['output'])
+        );
+        $this->assertSame(0, $result3['code']);
+    }
+
+    private static function executeCommand($command, $crashOnError)
+    {
+        $process = new Process($command, static::$fixtureDir);
+        $process->run();
+
+        $result = array(
+            'code' => $process->getExitCode(),
+            'output' => $process->getOutput(),
+            'stderr' => $process->getErrorOutput(),
+        );
+
+        if ($crashOnError && 0 !== $result['code']) {
+            throw new \RuntimeException(sprintf(
+                "Cannot execute `%s`:\n%s\nCode: %s\nExit text: %s\nError output: %s\nDetails:\n%s",
+                $command,
+                './'.static::$tmpFileName === $command
+                    ? implode('', array_map(function ($line) { return "$ $line"; }, file(static::$tmpFilePath)))."\n"
+                    : '',
+                $result['code'],
+                $process->getExitCodeText(),
+                $process->getErrorOutput(),
+                $result['output']
+            ));
+        }
+
+        return $result;
+    }
+
+    private static function executeScript(array $scriptParts, $crashOnError)
+    {
+        file_put_contents(static::$tmpFilePath, implode("\n", array_merge(array('#!/usr/bin/env bash', 'set -e', ''), $scriptParts)));
```

Comment:
to crash the script when any command of it is failing, not when last one will fail
we already do it in other `.sh` scripts we do have